### ISSUE TITLE
Enhancement to Handle.GET method

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#3038]
It is always not a good idea to keep string values inside code, especially when the values are presented more than one time.

Right now we do:

The handler

```cs
Handle.GET("/procurement/product/{?}", (string objectId) => {
...
});
```

The map

```cs
UriMapping.OntologyMap("/procurement/partials/product/@w", typeof(Product).FullName, null, null);
```

Things to improve: 

### Unify url format

We should not use different url format, `{?}` vs `@w`. Even me, after using this for a year, still sometimes forget where I should use which.

### Copy pasting url value

The url string value should be presented only once. Possible ways to improve:

```cs
var handler = Handle.GET("/procurement/product/{?}", (string objectId) => {
...
});

UriMapping.OntologyMap<Product>(handler);
```

```cs
Handle.GET("/procurement/product/{?}", (string objectId) => {
...
}).OntologyMap<Product>();
```

cc @alemoi, @dan31

**Comment by miyconst** [comment]
@warpech, @dan31, @alemoi, @per-samuelsson any input? My other issues got comments, but not this one.

**Comment by per-samuelsson** [comment]
> @per-samuelsson any input?

Sorry, didn't feel I had the right insight to comment on this one on that specific level. Just generally, I totally support this statement:

> We should not use different url format, `{?}` vs `@w`. Even me, after using this for a year, still sometimes forget where I should use which.

**Comment by warpech** [comment]
> We should not use different url format, {?} vs @w. Even me, after using this for a year, still sometimes forget where I should use which.

Totally agree!

> The url string value should be presented only once. Possible ways to improve:

Maps are meant to be written in a separate file, so I don't think this is applicable :(

**Comment by alemoi** [comment]
Sorry I missed this issue! Totally agree with unified URL format. Reason why I used processed URI is that in the beginning I was expecting other types of possible parameters not just last string. I also agree with @warperh about separation.

**Comment by warpech** [comment]
> Sorry I missed this issue! Totally agree with unified URL format. Reason why I used processed URI is that in the beginning I was expecting other types of possible parameters not just last string.

Can you estimate the effort to do this?

As always, the old syntax should be kept for backwards compatibility.

**Comment by alemoi** [comment]
The curly braces fix are now pushed to develop. Backwards compatible.
Here is an example, I pushed for Products app:
```
UriMapping.OntologyMap("/Products/partials/product/{?}", typeof(Product).FullName, null, null);
UriMapping.OntologyMap("/Products/partials/supplier/{?}", typeof(Organization).FullName, null, null);
UriMapping.Map("/Products/search?query={?}", UriMapping.MappingUriPrefix + "/search?query={?}");
```
Change log updated.

Additional change: Its no longer possible to register handlers with same signature. For example, one can't register handler "GET /{?}" with string parameter, and handler "GET /{?}" with integer parameter.

**Comment by miyconst** [comment]
Thanks @alemoi!

Did you add a deprecated warning for the old `@w` syntax?

**Comment by alemoi** [comment]
Could not deprecate that unfortunately, because its the same method signature as URI with curly braces.

**Comment by miyconst** [comment]
Okay, then leave it like this.

Eventually we will drop support for the old syntax, not now though.
